moxin-frontend/src/data/store.rs

```diff
@@ -1,11 +1,17 @@
 use crate::data::{chat::Chat, download::Download, search::Search};
+use anyhow::{Context, Result};
 use chrono::Utc;
 use makepad_widgets::DefaultNone;
 use moxin_backend::Backend;
 use moxin_protocol::data::{DownloadedFile, File, FileID, Model};
 use moxin_protocol::protocol::{Command, LoadModelOptions, LoadModelResponse};
 use std::collections::HashMap;
+use std::path::PathBuf;
 use std::sync::mpsc::channel;
+use std::{env, fs};
+
+const DEFAULT_DOWNLOADS_FOLDER: &str = "moxin/model_downloads";
```

Comment:
Don't we want to use a hidden folder instead, at least in MacOS. I see it would be more natural to have `.moxin` in the home folder. I don't remember what is the common practice in Win

Reply:
Good point, I guess we don't want to have users fiddle with the files in the folder as it might leave the db state inconsistent. 
I updated the code using `.moxin` and using a hidden attribute for directories on Windows, I tried the Windows separately from Moxin, I can't get Moxin to run on Windows easily, it's mostly WasmEdge and linking errors.

Either way, if this breaks for Windows whenever someone tries it, we can go back and fix it.

Reply:
Good point, I guess we don't want to have users fiddle with the files in the folder as it might leave the db state inconsistent. 

I updated the code using `.moxin`. For Windows, `.name` is a valid directory name but it won't make it hidden by default. To achieve changing visibility we'd need external crates that have APIs for modifying directory metadata. So I'm skipping that on Windows for now.

(I had pushed a commit earlier thinking I was able to do it without a crate, tried with and without a `winapi`, and got confused by the similar API names with `std`)